src/runtime/volatile/register.go

```diff
@@ -57,6 +57,16 @@ func (r *Register8) HasBits(value uint8) bool {
 	return (r.Get() & value) > 0
 }
 
+// ReplaceBits is a helper to simplify setting multiple bits high and/or low at
+// once. It is the volatile equivalent of:
+//
+//     r.Reg = (r.Reg & ^(mask << pos)) | value << pos
+//
+// go:inline
+func (r *Register8) ReplaceBits(value uint8, mask uint8, pos uint8) {
```

Comment:
If we want to add `ReplaceBits()` to the `volatile` package, seems to me it would be best done in a separate PR.

Reply:
https://github.com/tinygo-org/tinygo/pull/897